### DISCUSSION TITLE
node进程挂死

### DISCUSSION THREAD

**Discussion by seramat** [ddiu8081/chatgpt-demo#254] [Q&A] [answered]
我使用pnpm run dev & 之后正常访问, 但是过一段时间进程自动挂死. 请问log在哪里,如何查找原因.
另外, 有什么办法能让进程自动重启么?
谢谢

**Comment by ddiu8081** [comment, marked as answer]
不建议在正式使用的场合使用 `dev` 命令，这应该仅在开发阶段使用，不适合用来长时间运行。

要稳定部署，可以使用 `build` 命令打包 (#183)：

```
npm run build
node ./dist/server/entry.mjs
```

要实现自动重启，可使用 pm2 等工具运行 `./dist/server/entry.mjs`。

**Reply by ufoozhenghao** [reply]
> 不建议在正式使用的场合使用 `dev` 命令，这应该仅在开发阶段使用，不适合用来长时间运行。
> 
> 要稳定部署，可以使用 `build` 命令打包 (#183)：
> 
> ```
> npm run build
> node ./dist/server/entry.mjs
> ```
> 
> 要实现自动重启，可使用 pm2 等工具运行 `./dist/server/entry.mjs`。

请问该项目如何部署到自己的ECS呢？pm2 启动 entry.mjs 然后怎么弄？我看3000端口也没有服务器啊？之前没用过astro的框架，谢谢

**Reply by ddiu8081** [reply]
build命令打包后的产物在`dist`目录，可独立把`dist`目录上传至服务器。服务器需要配置 node 18+环境，使用 node 或 [pm2](https://github.com/Unitech/pm2) 启动 `./dist/server/entry.mjs` 即可在 3000 端口开放一个服务。

若外部访问 3000 不通，请检查云服务商是否有端口白名单机制。

**Reply by seramat** [reply]
> 不建议在正式使用的场合使用 `dev` 命令，这应该仅在开发阶段使用，不适合用来长时间运行。
> 
> 要稳定部署，可以使用 `build` 命令打包 (#183)：
> 
> ```
> npm run build
> node ./dist/server/entry.mjs
> ```
> 
> 要实现自动重启，可使用 pm2 等工具运行 `./dist/server/entry.mjs`。

谢谢, 已用pm2解决.

**Comment by ufoozhenghao** [comment]
谢谢 我试过了 node也是18+ 报一个 module错误，不知道咋回事 后来用docker可以了 谢谢回复

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Diu ***@***.***>
发送时间: Sunday, March 19, 2023 2:08:45 PM
收件人: ddiu8081/chatgpt-demo ***@***.***>
抄送: HaoXuan ***@***.***>; Comment ***@***.***>
主题: Re: [ddiu8081/chatgpt-demo] node进程挂死 (Discussion #254)


build命令打包后的产物在dist目录，可独立把dist目录上传至服务器。服务器需要配置 node 18+环境，使用 node 或 pm2<https://github.com/Unitech/pm2> 启动 ./dist/server/entry.mjs 即可在 3000 端口开放一个服务。

若外部访问 3000 不通，请检查云服务商是否有端口白名单机制。

―

**Comment by Pegessi** [comment]
请问我使用node执行entry.mjs后为什么前端会出现请求api/generate/ 500的错误呢
